common/lib/src/realm_types.dart

```diff
@@ -224,3 +225,223 @@ class RealmValue {
   @override
   String toString() => 'RealmValue($value)';
 }
+
+/// A base type for the supported geospatial shapes.
+sealed class GeoShape {}
+
+/// A point on the earth's surface.
+///
+/// It cannot be persisted as a property on a realm object.
+///
+/// Instead, you must use a custom embedded object with the following structure:
+/// ```dart
+/// @RealmModel(ObjectType.embeddedObject)
+/// class _Location {
+///   final String type = 'Point';
+///   List<double> coordinates = const [0, 0];
+/// }
+///
+/// You can then use it as a property on a realm object:
+///
+/// @RealmModel()
+/// class _Restaurant {
+///   @PrimaryKey()
+///   late String name;
+///   _Location? location;
+/// }
+/// ```
+
+class GeoPoint implements GeoShape {
+  final double lat;
+  final double lon;
+
+  /// Create a point from a [lat]itude and a [lon]gitude.
+  /// [lat] must be between -90 and 90, and [lon] must be between -180 and 180.
+  GeoPoint(this.lat, this.lon) {
+    if (lat < -90 || lat > 90) throw ArgumentError.value(lat, 'lat', 'must be between -90 and 90');
+    if (lon < -180 || lon > 180) throw ArgumentError.value(lon, 'lng', 'must be between -180 and 180');
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+    if (other is! GeoPoint) return false;
+    return lat == other.lat && lon == other.lon;
```

Comment:
This is a matter of preference, but can this not be rewritten like:

```suggestion
    return other is GeoPoint && lat == other.lat && lon == other.lon;
```

I think it's more readable than the alternating true/false/expression pattern in this and the other `==` overrides.

Reply:
It is common idiom. The two first lines are identical for all types, except of course for the type tested against. 
You really only have to _think_ about the lines after the first two.

I think I prefer as is, even though yours is one line shorter.